Modules/posixmodule.c

```diff
@@ -12808,6 +12815,7 @@ static PyMethodDef DirEntry_methods[] = {
     OS_DIRENTRY_STAT_METHODDEF
     OS_DIRENTRY_INODE_METHODDEF
     OS_DIRENTRY___FSPATH___METHODDEF
+    {"__class_getitem__", DirEntry_cls_getitem, METH_O|METH_CLASS, NULL},
```

Comment:
Would it be possible to use Argument Clinic for this (to be consistent with the rest of the code in this module)?

Reply:
I dont know if it possible, contextlib also uses argument clinic but they seperated `__class_getitem`. I thought there is a catch about this but I can try to do this with argument clinic.